packages/clay-css/src/scss/components/_empty-state.scss

```diff
@@ -0,0 +1,15 @@
+.empty-state-root {
```

Comment:
@pat270 should we be adding css for this or do you already have markup for the empty state?

Reply:
We don't have anything for empty state right now. Here are my suggestions:

Instead of relying on `h2` to size and space the text (e.g., No results found), we should scope it to a class like `empty-state-text`. This prevents styles from bleeing into the `empty-state` component if someone modifies `h2`.

Also the text for the `h2` element is limited to one line, you could wrap it in `<span class="text-truncate-inline"><span class="text-truncate"></span></span>` and make sure to add a `title` attribute with the full text.

We need to scope the message text inside `<div></div>` to a class `empty-state-message` and style it to match the design.

The image needs to be responsive just add class `img-fluid` (e.g., shrink to fit container size for narrow containers).

Empty State needs to have `max-width: 340px` and shrink to fit in narrow containers.

We don't delimit class names with double dash in Clay CSS (e.g., `.empty-state-root--no-image` should be `.empty-state-root-no-image`).

The way I would structure this component is:
```
<div class="empty-state empty-state-animation">
  <div class="empty-state-container">
    <div class="empty-state-image"></div>
    <div class="empty-state-text"></div>
    <div class="empty-state-message"></div>
    <div class="empty-state-actions"></div>
  </div>
</div>
```

Reply:
cc @kresimir-coko

Reply:
@pat270 Why introduce so many `div`s to the component?
I personally dislike having all that obsolete DOM. Putting 2 `spans` over the `h2` feels painful, let alone introducing 5, or so, new `divs`.
If that's the way we do things I'll do it that way, but it just feels wrong

Reply:
We already have this component in DXP, we shouldn't need to rewrite it entirely from scratch... can't we just port over what we have over there?

Reply:
@jbalsas I'm not sure how the porting process would look exactly, but I did something similar I assume. I copied the implementation from DXP and then made everything specific for Clay on top of that (stories, tests...)

Reply:
@kresimir-coko I wouldn't call it obsolete. They provide classes for devs to target if they need to modify it for some reason and prevent styles from bleeding into the component.

The spans for truncating text is the most failsafe way for truncating text I've found. It's pretty much a CSS hack. If there is a better way I'm always open to suggestions. Additional markup is a side effect of trying to be modular with our CSS components, it's a trade off. We can provide a class that truncates text in specific situations or we can provide two spans that always truncates text.

This PR doesn't fully implement the rules specified at https://liferay.design/lexicon/core-components/empty-states/. It also doesn't take into account the possibility of long text with no white space. For Empty Search Results, what if someone searches for ReallySuperInsanelyJustIncrediblyLongAndTotallyNotPossibleWordButWeAreReallyTryingToCoverAllOurBasesHereJustInCaseSomeoneIsNutsAsPerUsual? How will the text be displayed?

Reply:
You make a valid point @pat270 , I agree on the `spans`, it's the tool we have to achieve something, I don't know how those 2 spans work because this is the first time I'm seeing them, but I see their value. But on the topic of `divs` while yes, providing flexibility is awesome, and I haven't thought about that aspect as much as you have, I assumed that giving the `p` a class would be enough.

When it comes to the long text with no space, I haven't seen that in the spec.

I'll add the changes you advised and push when I get the chance :+1:

Reply:
@kresimir-coko long text no space is never in the spec, but always gets filed as a bug in every component. You can structure the markup how you want, the above was just a quick suggestion. We should have specific classes for most of elements though.